libs/sdk-core/src/error.rs

```diff
@@ -0,0 +1,30 @@
+use thiserror::Error;
+
+pub type SdkResult<T, E = SdkError> = Result<T, E>;
+
+/// Type of error returned by the SDK
+#[derive(Debug, Error)]
+pub enum SdkError {
+    #[error("Failed to connect to required services")]
+    ConnectFailed,
```

Comment:
Not sure we need ConnectFailed (sounds too generic). It seems that ConnectFailed is used in start up flow, so we can use InitFailed instead.
In other cases like in list_lsps we can use a more specific error (see my comment bellow).

Reply:
Done